### PR TITLE
Update ansible roles to support fedora

### LINKED ISSUE #228: Update the Ansible playbook
**Difficulty**: Easy to Medium

Update the CNDP [Ansible playbook](https://github.com/CloudNativeDataPlane/cndp/tree/main/ansible) to make sure it support for Fedora installs. 

The playbook needs to be updated to support libbpf and libxdp as the new libxdp contains the XSK APIs which were deprecated in libbpf , but libbpf is still required.

### PR DESCRIPTION
**Pull request by haoruan**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
Update ansible to support fedora, libbpf and libxdp installation.

#### Which issue(s) this PR fixes:
Fixes #228 

#### Special notes for your reviewer:
Update libbpf to v1.0.0, libxdp to v1.3.1
Tested in Ubuntu 20.04, Fedora 35, Fedora 38

Signed-off-by: Hao Ruan [hao.ruan@intel.com](mailto:hao.ruan@intel.com)
Signed-off-by: Xiaotian Chen [xiaotian.chen@intel.com](mailto:xiaotian.chen@intel.com)

### CODE REVIEW

**Comment by KeithWiles** [comment]
@maryamtahhan if you have the time can you review these changes for Fedora with ansible. If not it is OK, will try to figure it out.

**Comment by maryamtahhan** [comment]
@KeithWiles I will review

File: ansible/README.md [hunk 1]
```diff
@@ -23,12 +23,19 @@ EOF
...
 
+#### Ubuntu
 apt-get should now work to install the packages needed to use ansible.
 
 ```
 sudo apt update
 sudo apt-get install -y ansible
 ```
+#### Fedora
+apt-get should now work to install the packages needed to use ansible.
```
[the remark this answers is not in the document]
Comment (haoruan) [reply ansible/README.md]: Thanks, updated

File: ansible/README.md [hunk 2]
```diff
@@ -23,12 +23,19 @@ EOF
 
 ### dependencies
 
+#### Ubuntu
 apt-get should now work to install the packages needed to use ansible.
 
-```
+```bash
```
Comment (KeithWiles) [inline ansible/README.md:29]: Looks like a number of the image format sections ``` bash not correct. Some of them were in the file before being updated. Can you please fix these too.

Each ``` section needs to have a language type and I think it must have a space before the language type.

 Each code section needs to be something like this ``` bash with a space.

**Review by KeithWiles — CHANGES_REQUESTED** [review]
A few comments, but still reviewing the changes.

**Comment by KeithWiles** [comment]
In vscode I see a number of linting errors or warnings. The code seems to render OK, but I think they need to be fixed.

Missing blanks lines around a couple places some ``` blocks do not have a code type ‘``` bash’. I will fix them.

From: Maryam Tahhan ***@***.***>
Date: Wednesday, May 3, 2023 at 8:09 AM
To: CloudNativeDataPlane/cndp ***@***.***>
Cc: Wiles, Keith ***@***.***>, Mention ***@***.***>
Subject: Re: [CloudNativeDataPlane/cndp] Update ansible roles to support fedora (PR #296)

@maryamtahhan commented on this pull request.

________________________________

In ansible/README.md<https://github.com/CloudNativeDataPlane/cndp/pull/296#discussion_r1183664716>:

>  apt-get should now work to install the packages needed to use ansible.



-```

+```bash

@KeithWiles<https://github.com/KeithWiles> are you seeing some warnings? it renders correctly for me.

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#syntax-highlighting
https://www.thecodebuzz.com/highlight-bash-shell-code-in-markdown-readme-md-wiki-files/

File: ansible/README.md [hunk 3]
```diff
@@ -107,9 +114,10 @@ playbook
 
 ### Building CNDP
 
-After running Ansible to install all the dependencies, CNDP can be built by typing `make` in the
+After running Ansible to install all the dependencies, please set `PKG_CONFIG_PATH`, then CNDP can be built by typing `make` in the
 top level dir:
 
 ```
+export PKG_CONFIG_PATH=/usr/lib/pkgconfig
 make
```
[the remark this answers is not in the document]
Comment (haoruan) [reply ansible/README.md]: Updated

[the remark this answers is not in the document]
Comment (KeithWiles) [reply ansible/README.md:123]: Does not look like this was converted to `make rebuild-install` and is still using make only command.

File: ansible/cndp-ansible/roles/setup_hugepages/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,3 @@
+update_grub_cmd:
```
Comment (KeithWiles) [inline ansible/cndp-ansible/roles/setup_hugepages/defaults/main.yml:1]: Copyright notice.

File: ansible/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (haoruan) [reply ansible/README.md]: Sorry, it should be updated now

File: ansible/cndp-ansible/roles/install_libxdp/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,79 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2020-2023 Intel Corporation
```
[the remark this answers is not in the document]
Comment (haoruan) [reply ansible/cndp-ansible/roles/install_libxdp/tasks/main.yml]: Correct it to 2023

**Pull request merged**